database/schema.sql

```diff
@@ -0,0 +1,181 @@
+
+-- Creating 'Assignments' table.
+--
+create table if not exists Assignments (
+    hash       BLOB     not null,
+    course_id  INTEGER  not null,
+    type       INTEGER  not null,
+    desc       TEXT     null,
+
+    primary key (hash),
+    foreign key (course_id) references Courses(id)
+
+) without rowid;
+
+create index if not exists Assigments_course_id on Assignments (course_id);
+
+-- Creating 'StudentAssignments' table.
+--
+-- [Appears to be] many-to-many ref between Students and Assigments.
+--
+create table if not exists StudentAssignments (
+    student_addr     BLOB  not null,
+    assignment_hash  BLOB  not null,
+
+    primary key (student_addr, assignment_hash),
+
+    foreign key (student_addr)    references Students    (addr)
+    foreign key (assignment_hash) references Assignments (hash)
+
+) without rowid;
+
+create index if not exists StudentAssigments_student_addr    on StudentAssignments (student_addr);
+create index if not exists StudentAssigments_assignment_hash on StudentAssignments (assignment_hash);
+
+-- Creating 'Submissions' table.
+--
+create table if not exists Submissions (
+    hash             BLOB     not null,
+    student_addr     BLOB     not null,
+    assignment_hash  BLOB     not null,
+    type             INTEGER  not null,
+    signature        BLOB     not null,
+
+    primary key (hash),
+    foreign key (student_addr)    references Students    (addr),
+    foreign key (assignment_hash) references Assignments (hash),
+
+    -- Composite foreign key.
+    -- We may not need it, actually, but it may help us.
+    --
+    foreign key (student_addr, assignment_hash)
+        references StudentAssignments(student_addr, assignment_hash)
+
+) without rowid;
+
+create index if not exists Submissions_student_addr    on Submissions (student_addr);
+create index if not exists Submissions_assignment_hash on Submissions (assignment_hash);
+
+-- Creating 'Transactions' table.
+--
+create table if not exists Transactions (
+    hash             BLOB     not null,
+    submission_hash  BLOB     not null,
+    grade            INTEGER  not null,
+    time             TIME     not null,
+
+    primary key (hash),
+    foreign key (submission_hash) references Submissions(hash)
+
+) without rowid;
+
+create index if not exists Transactions_submission_hash on Transactions (submission_hash);
+
+-- Creating 'Blocks' table.
+--
+create table if not exists Blocks (
+    hash       BLOB  not null,
+    time       TIME  not null,
+    prev_hash  BLOB  null,
+    atg_delta  BLOB  not null,
+    mroot      BLOB  not null,
+    mtree      BLOB  not null,
+
+    primary key (hash),
+    foreign key (prev_hash) references Blocks(hash)
+
+) without rowid;
+
+create index if not exists Blocks_prev_hash on Blocks (prev_hash);
+
+-- Creating 'Educator' table.
+--
+create table if not exists Educator (
+    signature  BLOB non null,
+
+    primary key (signature)
+) without rowid;
```

Comment:
What's this table for, which signatures are stored there?

Reply:
Its a table to store current `Educator` in use. I thought about storing its signature in the database; however now I see that some keyring storage is better for that.

Reply:
Exactly, we should store keys in a separate keyfile.